optimum/neuron/modeling_decoder.py

```diff
@@ -0,0 +1,208 @@
+        - generation_config ([`~transformers.GenerationConfig`]) -- The generation configuration used by default when calling `generate()`.
+    """
+
+    CHECKPOINT_DIR = "checkpoint"
+    COMPILED_DIR = "compiled"
+
+    def __init__(
+        self,
+        model: torch.nn.Module,
+        config: "PretrainedConfig",
+        model_path: Union[str, Path, TemporaryDirectory],
+        generation_config: Optional[GenerationConfig] = None,
+    ):
+        if not is_transformers_neuronx_available() or not isinstance(model, NeuronxPretrainedModel):
+            raise ValueError("The source model must be a transformers_neuronx.PreTrainedModel.")
+
+        super().__init__(model, config)
+        self.model_path = model_path
+        if generation_config is None:
+            generation_config = GenerationConfig.from_model_config(config)
+        self.generation_config = generation_config
+        self.device = torch.device("cpu")
+
+    @classmethod
+    def _from_transformers(
+        cls,
+        model_id: str,
+        config: "PretrainedConfig",
+        use_auth_token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        force_download: bool = False,
+        cache_dir: Optional[str] = None,
+        subfolder: str = "",
+        local_files_only: bool = False,
+        trust_remote_code: bool = False,
+        task: Optional[str] = None,
+        **kwargs,
+    ) -> "NeuronDecoderModel":
+        if not is_transformers_neuronx_available():
+            raise ModuleNotFoundError("The transformers_neuronx package is required to export the model.")
+
+        if task is None:
+            task = TasksManager.infer_task_from_model(cls.auto_model_class)
+
+        # Instantiate the exporter for the specified configuration and task
+        exporter = get_exporter(config, task)
+
+        # Split kwargs between model and neuron args
+        model_kwargs, neuron_kwargs = exporter.split_kwargs(**kwargs)
+
+        # Instantiate the transformers model checkpoint
+        model = TasksManager.get_model_from_task(
+            task=task,
+            model_name_or_path=model_id,
+            subfolder=subfolder,
+            revision=revision,
+            framework="pt",
+            cache_dir=cache_dir,
+            use_auth_token=use_auth_token,
+            local_files_only=local_files_only,
+            force_download=force_download,
+            trust_remote_code=trust_remote_code,
+            **model_kwargs,
+        )
+
+        # Save the model checkpoint in a temporary directory
+        checkpoint_dir = TemporaryDirectory()
+        save_pretrained_split(model, checkpoint_dir.name)
+
+        # Update the config
+        config.neuron = {"task": task, "neuron_kwargs": neuron_kwargs}
+
+        return cls._from_pretrained(checkpoint_dir, config)
+
+    @classmethod
+    def _get_neuron_paths(cls, model_dir: Union[str, Path, TemporaryDirectory]) -> Tuple[str, str, str]:
```

Comment:
What if I want to load from the hub, is it something feasible?

https://github.com/huggingface/optimum-neuron/blob/980dc5cfaa15b23ac3691b78f640a7182e3fd4df/optimum/neuron/modeling_base.py#L141-L143

Reply:
I don't know, because my knowledge of the interactions with the hub is still very limited. This is something that would come in subsequent pull-requests (I might need guidance).

Reply:
Never mind, we can get this in first and add the hub support in a separate PR. 
Basically it is just done through the `HfApi()` of `huggingface_hub`, as how it was done for other optimum modeling: https://github.com/huggingface/optimum-neuron/blob/980dc5cfaa15b23ac3691b78f640a7182e3fd4df/optimum/neuron/modeling_base.py#L141-L148

Feel free to ping me when you start to tackle this.